metpy/calc/thermo.py

```diff
@@ -1038,3 +1038,165 @@ def most_unstable_parcel(p, temperature, dewpoint, heights=None,
+    tmpk_out : bool, optional
+        If true, will calculate temperature and output as the last item in the output list.
+        Defaults to False.
+    max_iters : int, optional
+        The maximum number of iterations to use in calculation, defaults to 50.
+    eps : float, optional
+        The desired absolute error in the calculated value, defaults to 1e-6.
+
+    Notes
+    -----
+    Input variable arrays must have the same number of vertical levels as the pressure levels
+    array. Pressure is calculated on isentropic surfaces by assuming that temperature varies
+    linearly with the natural log of pressure. Linear interpolation is then used in the
+    vertical to find the pressure at each isentropic level. Interpolation method from
+    [Hoerling1993]_. Any additional arguments are assumed to vary linearly with temperature
+    and will be linearly interpolated to the new isentropic levels.
+
+    See Also
+    --------
+    potential_temperature
+
+    """
+    # iteration function to be used later
+    def _isen_iter(pk1, isentlevs2, ka, a, b, pok):
+        ekp = np.exp((-ka) * pk1)
+        t = a * pk1 + b
+        f = isentlevs2 - pok * t * ekp
+        fp = pok * ekp * ((ka) * t - a)
+        return pk1 - (f / fp)
+
+    # Change when Python 2.7 no longer supported
+    # Pull out keyword arguments
+    tmpk_out = kwargs.pop('tmpk_out', False)
+    max_iters = kwargs.pop('max_iters', 50)
+    eps = kwargs.pop('eps', 1e-6)
+    axis = kwargs.pop('axis', 0)
+
+    # Get dimensions in temperature
+    ndim = temperature.ndim
+
+    # Convert units
+    pres = pressure.to('hPa')
+    temperature = temperature.to('kelvin')
+
+    # Make pressure the same shape as temperature
+    for dim in range(ndim):
+        if dim < axis:
+            pres = np.repeat(pres[np.newaxis, ...], temperature.shape[dim], axis=0)
+        elif dim > axis:
+            pres = np.repeat(pres[..., np.newaxis], temperature.shape[dim], axis=-1)
+
+    # Sort input data
+    sort_pres = np.argsort(pres.m, axis=axis)
+    sort_pres = np.swapaxes(np.swapaxes(sort_pres, 0, axis)[::-1], 0, axis)
+    sorter = broadcast_indices(pres, sort_pres, ndim, axis)
+    levs = pres[sorter]
+    theta_levels = np.asanyarray(theta_levels.to('kelvin')).reshape(-1)
+    sort_isentlevs = np.argsort(theta_levels)
+    tmpk = temperature[sorter]
+    isentlevels = theta_levels[sort_isentlevs]
+
+    # Make the desired isentropic levels the same shape as temperature
+    isentlevs2 = isentlevels
+    for dim in range(ndim):
+        if dim < axis:
+            isentlevs2 = np.repeat(isentlevs2[np.newaxis, ...], temperature.shape[dim], axis=0)
+        elif dim > axis:
+            isentlevs2 = np.repeat(isentlevs2[..., np.newaxis], temperature.shape[dim],
+                                   axis=-1)
+
+    # exponent to Poisson's Equation, which is imported above
+    ka = kappa.to('(J*s^2)/(kg*m^2)')
```

Comment:
const.kappa is natively in units of j/g*s^2/m^2. We need it in terms of kg for the calculations later since units have to be removed.